File: Examples/VisualBasic/ModifyingAndConvertingImages/ConcatTIFFImages.vb
Imports System.IO
Imports Aspose.Imaging.FileFormats.Tiff

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https://Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages
    Public Class ConcatTIFFImages
        Public Shared Sub Run()
            ' To get proper output please apply a valid Aspose.Imaging License. You can purchase full license or get 30 day temporary license from http:// Www.aspose.com/purchase/default.aspx.")
            ' ExStart:ConcatTIFFImages
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_ModifyingAndConvertingImages()

            ' Create a copy of original image to avoid any alteration
            File.Copy(dataDir & Convert.ToString("demo.tif"), dataDir & Convert.ToString("TestDemo.tif"), True)

            ' Create an instance of TiffImage and load the copied destination image
            Using image__1 As TiffImage = DirectCast(Image.Load(dataDir & Convert.ToString("TestDemo.tif")), TiffImage)
                ' Create an instance of TiffImage and load the source image
                Using image1 As TiffImage = DirectCast(Image.Load(dataDir & Convert.ToString("sample.tif")), TiffImage)
                    ' Create an instance of TIffFrame and copy active frame of source image
                    Dim frame As TiffFrame = TiffFrame.CopyFrame(image1.ActiveFrame)

                    ' Add copied frame to destination image
                    image__1.AddFrame(frame)

                    ' Save the image with changes.
                    image__1.Save(dataDir & Convert.ToString("ConcatTIFFImages_out.tiff"))
                End Using
            End Using
            ' ExEnd:ConcatTIFFImages
            ' Display Status.
            Console.WriteLine("Concatenation of TIF files done successfully.")
        End Sub
    End Class
End Namespace

File: Examples/CSharp/ModifyingAndConvertingImages/ConcatTIFFImages.cs
using System;
using System.IO;
using Aspose.Imaging.FileFormats.Tiff;

/*
This project uses the Automatic Package Restore feature of NuGet to resolve the Aspose.Imaging for .NET API reference 
when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from https://releases.aspose.com/, 
install it, and then add its reference to this project. For any issues, questions, or suggestions, 
please feel free to contact us using https://forum.aspose.com/
*/

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages
{
    public class ConcatTIFFImages
    {
        public static void Run()
        {
            Console.WriteLine("Running example ConcatTIFFImages");
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_ModifyingAndConvertingImages();

            // Create a copy of the original image to avoid any alteration.
            File.Copy(dataDir + "demo.tif", dataDir + "TestDemo.tif", true);

            // Load the copied destination image.
            using (TiffImage image = (TiffImage)Image.Load(dataDir + "TestDemo.tif"))
            {
                // Load the source image.
                using (TiffImage image1 = (TiffImage)Image.Load(dataDir + "sample.tif"))
                {
                    // Copy the active frame of the source image, add the copied frame to the destination image, and save the result.
                    TiffFrame frame = TiffFrame.CopyFrame(image1.ActiveFrame);
                    image.AddFrame(frame);
                    image.Save(dataDir + "ConcatTIFFImages_out.tiff");
                }
            }

            Console.WriteLine("Finished example ConcatTIFFImages");
        }
    }
}